src/viking.js

```diff
@@ -1,14 +1,113 @@
 // Soldier
-class Soldier {}
+class Soldier {
+  constructor(health, strength){
+    this.health=health
+    this.strength=strength
+  }
+
+  attack(){
+    return this.strength;
+  }
+
+  receiveDamage(damage){
+    this.health=this.health-damage
+  }
+}
+
+/*const soldier = new Soldier(20,20)
+console.log(soldier)
+console.log(soldier.receiveDamage(15),soldier)*/
+
 
 // Viking
-class Viking {}
+class Viking extends Soldier {
+  constructor(name,health,strength){
+    super(health,strength);
+    this.name=name;
+  }
+
+  receiveDamage(damage){
+    this.health=this.health-damage
+    if (this.health>0){
+      return `${this.name} has received ${damage} points of damage`
+    } else return `${this.name} has died in act of combat`
+  }
+
+  battleCry(){
+    return `Odin Owns You All!`
+  }
+}
+
+/*const viking= new Viking(`gian`,20,25)
+console.log(viking)
+console.log(viking.receiveDamage(10),viking)*/
 
 // Saxon
-class Saxon {}
+class Saxon extends Soldier{
+  
+
+  receiveDamage(damage){
+    this.health=this.health-damage
+    if (this.health>0){
+      return `A Saxon has received ${damage} points of damage`
+    } else return `A Saxon has died in combat`
+  }
+}
+
+/*const saxon = new Saxon(20,25)
+console.log(saxon)
+console.log(saxon.receiveDamage(29),saxon)*/
 
 // War
-class War {}
+class War {
+  constructor(){
+  this.vikingArmy= [];
+  this.saxonArmy=[];
+  }
+
+  addViking(viking){
+    this.vikingArmy.push(viking)
+  }
+  addSaxon(saxon){
+    this.saxonArmy.push(saxon)
+  }
+  vikingAttack(){
+    let randomSaxon = Math.floor(Math.random() * (this.saxonArmy.length));
+    let randomViking = Math.floor(Math.random() * (this.vikingArmy.length));
+    let result = this.saxonArmy[randomSaxon].receiveDamage(this.vikingArmy[randomViking].strength);
```

Comment:
Está bien cómo lo hiciste. Te pongo otra solución con un filter para que tengas otra visión!!
` vikingAttack() {
const viking = randomElement(this.vikingArmy);
const saxon = randomElement(this.saxonArmy);

const attackResult = saxon.receiveDamage(viking.strength);
this.saxonArmy = this.saxonArmy.filter(saxon => saxon.health > 0);
return attackResult;


}`